Miscellaneous heteroatoms fixes and additions

alongd opened the pull request:
This PR adds:
- A rate for `H2S + O2 <=> SH + HO2` to primarySulfurLibrary
- Thermo value correction for `OSO(T)` in the primaryNS library (`OOS(T)` was also updated)
- A sublibrary for GlarborgH2S
- AdjList updates in thermo libraries

alongd commented:
@mjohnson541, this is ready for review

mjohnson541 commented on input/kinetics/families/intra_NO2_ONO_conversion/training/reactions.py:
```diff
@@ -7,3 +7,11 @@
+    label = "C2H5NO2 <=> CH3CH2ONO",
+    degeneracy = 1,
+    kinetics = Arrhenius(A=(2.1e+10, 's^-1'), n=1, Ea=(60660, 'cal/mol'), T0=(1, 'K')),
+    shortDesc = u"""From kinetic library NOx2018""",
```
Can you provide a better description of how this reaction rate was determined?

mjohnson541 commented on input/kinetics/families/Intra_R_Add_Endocyclic/groups.py:
```diff
@@ -33,8 +33,8 @@
-1 *2 [Cd,Ct,Cb,Cbf,CO,CS,Cdd,N,S4d,S6d,S6dd] u0 {2,[D,T,B]}
-2 *3 [Cd,Ct,Cb,Cbf,O2d,S2d,Cdd,N]            u0 c0 {1,[D,T,B]}
+1 *2 [Cd,Ct,Cb,Cbf,CO,CS,Cdd,N,O,S] u0 {2,[D,T,B]}
+2 *3 [Cd,Ct,Cb,Cbf,Cdd,N,O,S]       u0 c0 {1,[D,T,B]}
```
I understand the change in *3, and I'm unsure about the S chemistry, but can *2 really be O?  Doesn't that force it to have three bonds initially?

alongd replied:
I believe O should be allowed in *2. It should be either `O4dc` or `O4tc`, I'll change the atomTypes on this line to be more specific.

alongd commented:
@mjohnson541, let me know whether I can squash the two fixups.
We'll look at the tests before merging (RMG-tests seem unstable, I restarted it)

mjohnson541 commented:
If we can solve the RMG-tests issues I'm ok with merging this.

alongd commented:
@mjohnson541, I removed the one sulfur reaction I added to the primarySulfurLibrary. I plan to add it later with additional improvements, hopefully it will make sense then. Now the rmg-tests don't complain. I think this branch is finally good to go.

alongd commented:
@mjohnson541, is this ready to merge?
